### PR TITLE
feat: support windows , run in multiple networks and support heartbeat

### PR DESCRIPTION
**Pull request by timandy**
Hi, there.
I made some features, hope can help you.

1. Support compile projects on windows by goc.
2. Fix goc client can not get the correct client-ip when the client has multiple networks, such as run in docker.
3. The goc-client support register self to goc-server every 10 second. And the goc-sever kick out clients which is timeout (no register over 20 seconds) every 5 second.

### CODE REVIEW

**Comment by timandy** [comment]
test case failed caused by server write log not flushed to disk. The ci in my fork repo is successed.

File: pkg/cover/instrument.go [hunk 1]
```diff
@@ -322,64 +338,49 @@ func isNetworkError(err error) bool {
...
-	var nonLocalIPV4 string
-	for _, addr := range adds {
-		if ipNet, ok := addr.(*net.IPNet); ok && ipNet.IP.To4() != nil {
-			if ipNet.IP.IsLoopback() {
-				localIPV4 = ipNet.IP.String()
-			} else {
-				nonLocalIPV4 = ipNet.IP.String()
-			}
-		}
+func getRealHost(port int) (host string, err error) {
```
Comment (CarlJi) [inline pkg/cover/instrument.go:371]: `host` and `err` are not in use.

File: pkg/cover/instrument.go [hunk 2]
```diff
@@ -416,7 +417,46 @@ func genProfileAddr(profileAddr string) {
 	}
 	defer f.Close()
 
-	fmt.Fprintf(f, strings.TrimPrefix(profileAddr, "http://"))
+	host, err := getRealHost(port)
+	if err != nil {
```
Comment (CarlJi) [inline pkg/cover/instrument.go:421]: this err isn't caught. we'd better print it to let outside knows.

**Comment by CarlJi** [comment]
Hi @timandy, Thanks for your contributions. This seems a little big changes and we need more verifications especially for the new adapters on windows & k8s env.

Also, i am thinking whether it is necessary or not for the heartbeats enhancement, since now `goc profile` will ignore the failed registered services by default. Would you please provide more explanations for your thoughts about this feature?

File: pkg/cover/instrument.go (hunk 1 above)
Comment (timandy) [reply pkg/cover/instrument.go:371]: @CarlJi  fixed

**Comment by timandy** [comment]
> Also, i am thinking whether it is necessary or not for the heartbeats enhancement, since now `goc profile` will ignore the failed registered services by default. Would you please provide more explanations for your thoughts about this feature?

@CarlJi  Thanks for your review.  Our service is deployed in the container. After several restarts, a lot of invalid clients will be generated. Although the goc server can filter invalid clients, these operations need network probing, which affects performance.

**Pull request closed without merging**